### ISSUE TITLE
Warning when commanding scene

### ISSUE THREAD

**Issue by crxporter** [TD22057/insteon-mqtt#156]
I'm working on setting up my scenes. I thought I had everything working properly so I dove in. I had 2 scenes working, the mqtt messages were coming through, everything looked great.

Then I tried to run a scene with about 30 of my switches. Sent all of the commands to add responders, on levels, etc. The scene itself is working properly but I'm not getting the mqtt messages for the devices being triggered.

So I dialed it back and went back to a 1-device scene then checked `sudo systemctl status insteon-mqtt.service` and saw that I was getting a warning each time I tried to set a scene. I went back to the other scenes and noticed I am getting the same warning.

Status output when triggering the scene:
```
Jul 01 17:05:37 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:37 INFO Modem: Modem input command: {'cmd': 'on', 'group': 99}
Jul 01 17:05:37 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:37 INFO Modem: Modem scene 99 on=on
Jul 01 17:05:37 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:37 INFO Protocol: Write message to modem: Modem Scene: grp: 99 cmd: 0x11 0x00
Jul 01 17:05:37 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:37 INFO Protocol: Read 0x61: Modem Scene: grp: 99 cmd: 0x11 0x00 ack: True
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 INFO Protocol: Read 0x58: All link status ack: 1
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 INFO Modem: 4c.18.14 (modem) broadcast to 4b.b3.5f for group 99
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 INFO Switch: Setting device 4b.b3.5f (mirror) on True normal scene
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 INFO Switch: MQTT received on/off 4b.b3.5f (mirror) on: True normal 'scene'
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 INFO Protocol: Read 0x50: Std: 4b.b3.5f Type.CLEANUP_ACK grp: 63 cmd: 11 63
Jul 01 17:05:38 pinsteon insteon-mqtt[499]: 2019-07-01 17:05:38 WARNING Protocol: No read handler found for message type 0x50: Std: 4b.b3.5f Type.CLEANUP_ACK grp: 63 cmd: 11 63
```

It seems I'm getting the warning `No read handler found for message type 0x50` for every device in every scene.

Sorry if I've missed an obvious step - I've tried to read through everything. Thanks for your time!

Hardware: 2477S and 2477D. The messages here are for a switch. I also have a few micro-dimmers which I haven't tried adding to scenes yet, but they have historically always acted just like the wall dimmer switches. I'm running on a USB PLM connected to a pi 3B+ with nothing else running on the pi.

**Comment by crxporter** [comment]
I am noticing another issue that could be related here. If I set a scene where switch A is on and switch B is off then send the `on` command to that scene - both switches get mqtt commands back showing `on` and reason `scene` (running dev branch).

The lights themselves are actually changing properly - switch A is on and switch B is off. However the mqtt messages are not matching what the switches are actually doing. Sending a `refresh_all` command updates properly...

I have plans for scenes where most of the lights turn off and some turn on...

**Comment by TD22057** [comment]
Please turn on debug level logging in the config file and post what you see there.  It's easier to just look the log file rather than use systemctl.

For your second issue, please check the log carefully when doing the refresh.  The only way I know of for them to be out of sync is if the database stored by insteon-mqtt is not up to date.  And I don't know what you mean by "not matching".  What mqtt messages were posted (run `mosquitto_sub -v -t 'insteon/#'`)? Are they not being sent?  Or are they being sent w/ the wrong reason?  Wrong state?

**Comment by crxporter** [comment]
> Are they not being sent? Or are they being sent w/ the wrong reason? Wrong state?

It seems like the messages are being sent - but with the wrong state. For example when the scene sets a dimmer to 40% the MQTT message shows it being set to 100%.

Or if the scene "on" sets a switch (not dimmer) to `OFF` then the MQTT message actually shows it turning `ON`

I haven't had a chance to look at the logs for the other issue, I'll hopefully have some time later this week.

**Comment by crxporter** [comment]
I've set my log level to debug and copied the output when commanding a scene.

Turning two switches on with scene 100:
```
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 INFO Mqtt: MQTT message insteon/command/modem b'{"cmd":"scene","group":100,"is_on":1}'
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 UI Mqtt: Commanding Modem device 4c.18.14 (modem) cmd=scene
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 INFO Modem: Modem scene 100 on=on
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 INFO Protocol: Write message to modem: Modem Scene: grp: 100 cmd: 0x11 0x00
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG Protocol: Write bytes to modem: b'\x02ad\x11\x00'
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG Serial: Wrote 5 bytes to serial /dev/ttyUSB0
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 INFO Protocol: Read 0x61: Modem Scene: grp: 100 cmd: 0x11 0x00 ack: True
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 INFO Protocol: Read 0x50: Std: 34.0a.ec Type.CLEANUP_ACK grp: 64 cmd: 11 64
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG Protocol: Setting next write time: 1562786712.717630
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG MsgHistory: Received 0 hops, total 0 for 2 entries
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:12 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:12 WARNING Protocol: No read handler found for message type 0x50: Std: 34.0a.ec Type.CLEANUP_ACK grp: 64 cmd: 11 64
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Protocol: Read 0x58: All link status ack: 1
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG ModemScene: Modem scene 100 command ACK
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Modem: Found 2 responders in group 100
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Modem: Group 100 -> ['34.0a.ec', '41.d1.44']
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Modem: 4c.18.14 (modem) broadcast to 34.0a.ec for group 100
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Switch: Setting device 34.0a.ec (backporch) on True normal scene
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Switch: MQTT received on/off 34.0a.ec (backporch) on: True normal 'scene'
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Mqtt: MQTT publish insteon/34.0a.ec/state { "state" : "ON", "reason" : "scene", "mode" : "normal" } qos=1 ret=1
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Modem: 4c.18.14 (modem) broadcast to 41.d1.44 for group 100
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Switch: Setting device 41.d1.44 (frontporch) on True normal scene
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Switch: MQTT received on/off 41.d1.44 (frontporch) on: True normal 'scene'
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Mqtt: MQTT publish insteon/41.d1.44/state { "state" : "ON", "reason" : "scene", "mode" : "normal" } qos=1 ret=1
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 UI Mqtt: Scene command complete
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Protocol: Write handler finished
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 INFO Protocol: Read 0x50: Std: 41.d1.44 Type.CLEANUP_ACK grp: 64 cmd: 11 64
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Protocol: Setting next write time: 1562786713.484101
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG MsgHistory: Received 0 hops, total 1 for 2 entries
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 WARNING Protocol: No read handler found for message type 0x50: Std: 41.d1.44 Type.CLEANUP_ACK grp: 64 cmd: 11 64
Jul 10 14:25:13 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:13 DEBUG Mqtt: MQTT writing
```

Turning two switches off with the same scene:
```
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Mqtt: MQTT message insteon/command/modem b'{"cmd":"scene","group":100,"is_on":0}'
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 UI Mqtt: Commanding Modem device 4c.18.14 (modem) cmd=scene
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Modem: Modem scene 100 on=off
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Protocol: Write message to modem: Modem Scene: grp: 100 cmd: 0x13 0x00
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Write bytes to modem: b'\x02ad\x13\x00'
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Serial: Wrote 5 bytes to serial /dev/ttyUSB0
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Protocol: Read 0x61: Modem Scene: grp: 100 cmd: 0x13 0x00 ack: True
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Protocol: Read 0x50: Std: 34.0a.ec Type.CLEANUP_ACK grp: 64 cmd: 13 64
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Setting next write time: 1562786717.548983
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG MsgHistory: Received 0 hops, total 0 for 3 entries
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 WARNING Protocol: No read handler found for message type 0x50: Std: 34.0a.ec Type.CLEANUP_ACK grp: 64 cmd: 13 64
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Protocol: Read 0x58: All link status ack: 1
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Passing msg to write handler: ModemScene handler
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG ModemScene: Modem scene 100 command ACK
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Modem: Found 2 responders in group 100
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Modem: Group 100 -> ['34.0a.ec', '41.d1.44']
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Modem: 4c.18.14 (modem) broadcast to 34.0a.ec for group 100
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Switch: Setting device 34.0a.ec (backporch) on False normal scene
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Switch: MQTT received on/off 34.0a.ec (backporch) on: False normal 'scene'
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Mqtt: MQTT publish insteon/34.0a.ec/state { "state" : "OFF", "reason" : "scene", "mode" : "normal" } qos=1 ret=1
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Modem: 4c.18.14 (modem) broadcast to 41.d1.44 for group 100
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Switch: Setting device 41.d1.44 (frontporch) on False normal scene
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Switch: MQTT received on/off 41.d1.44 (frontporch) on: False normal 'scene'
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Mqtt: MQTT publish insteon/41.d1.44/state { "state" : "OFF", "reason" : "scene", "mode" : "normal" } qos=1 ret=1
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 UI Mqtt: Scene command complete
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Write handler finished
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Mqtt: MQTT writing
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 INFO Protocol: Read 0x50: Std: 41.d1.44 Type.CLEANUP_ACK grp: 64 cmd: 13 64
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG Protocol: Setting next write time: 1562786717.852779
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 DEBUG MsgHistory: Received 0 hops, total 1 for 3 entries
Jul 10 14:25:17 pi4 insteon-mqtt[10365]: 2019-07-10 14:25:17 WARNING Protocol: No read handler found for message type 0x50: Std: 41.d1.44 Type.CLEANUP_ACK grp: 64 cmd: 13 64
```

These commands work and the MQTT messages being sent are correct. I do still see the warning of "No read handler". Trying to start simple and work into the more complex scenes which I can build once I've figured things out a little better...

**Comment by TD22057** [comment]
FYI I don't think the warning is important.  Insteon can send multiple messages out for cleanup sometimes.  See the 4'th line from the bottom of the last log - it gets a cleanup and processes it.  Then another cleanup arrives so that's ignored.  I probably won't try and figure out a nice way to suppress the warning since in other cases it's useful.

I'll have to look closer at your logs to see what's going on w/ the dimmer issue.

**Comment by krkeegan** [comment]
I also agree about the 0x50 warning messages.  Those are additional cleanup status messages that we currently ignore.  Those warnings are just a red herring.

**Comment by TD22057** [comment]
FYI - I just set 2 buttons a KeypadLinc to be connected to a dimmer.  Button 4 was linked to the dimmer at 50% brightness and Button 5 (on) was linked to the dimmer (off).  Both appear to work fine (after doing a refresh to get the local db to match the device).  If I turn the dimmer on and click KP button 5, it turns on and the dimmer turns off.  If I click KP button 4, the dimmer goes to 50%.  The MQTT messages are correct in both cases.

**Comment by TD22057** [comment]
@crxporter I'm not sure what I can do to help.  In your long post (2 on, 2 off), if I'm understand what you're saying, then look correct right?  Both devices report on and off.  In my local test, I get the right results even from mixed on/off results.

When you set up the problem case (1 on, 1 off, I think), can you post the database dump for the modem and each device, and the log messages when the scene is triggered?

**Comment by crxporter** [comment]
I honestly don’t even remember the issue completely anymore. In the end I got this plugin working just enough to hook it up to homekit in nodered and we’ve been using Siri to do everything. 

My Insteon setup is getting retired soon so I won’t really be putting in anymore effort to get things fixed. I probably won’t even update to the new yaml scenes setup, which I was originally very excited about when I came to this project. 

It’s not you - it’s Insteon. I’ve got some devices on the fritz (PLM and others) and have decided to give zwave a try. Thanks for the support and best of luck with your Insteon futures!
